commit bbefd4fc8f590e765b455ef0a4deb6c105fee305
Author: Paolo Bonzini <pbonzini@redhat.com>
Date:   Mon Dec 14 07:49:54 2020 -0500

    KVM: x86: move kvm_inject_gp up from kvm_set_xcr to callers
    
    Push the injection of #GP up to the callers, so that they can just use
    kvm_complete_insn_gp.
    
    Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>

diff --git a/arch/x86/kvm/vmx/vmx.c b/arch/x86/kvm/vmx/vmx.c
index 0cd24ae2db08..d995f47a08ef 100644
--- a/arch/x86/kvm/vmx/vmx.c
+++ b/arch/x86/kvm/vmx/vmx.c
@@ -5223,9 +5223,8 @@ static int handle_xsetbv(struct kvm_vcpu *vcpu)
 	u64 new_bv = kvm_read_edx_eax(vcpu);
 	u32 index = kvm_rcx_read(vcpu);
 
-	if (kvm_set_xcr(vcpu, index, new_bv) == 0)
-		return kvm_skip_emulated_instruction(vcpu);
-	return 1;
+	int err = kvm_set_xcr(vcpu, index, new_bv);
+	return kvm_complete_insn_gp(vcpu, err);
 }
 
 static int handle_apic_access(struct kvm_vcpu *vcpu)